tests/test_ace.rb

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'ciscotest'
+require_relative '../lib/cisco_node_utils/acl'
+require_relative '../lib/cisco_node_utils/ace'
+
+# TestX__CLASS_NAME__X - Minitest for X__CLASS_NAME__X node utility class
+class TestAceV4 < CiscoTestCase
+  attr_reader :acl_name, :seqno, :afi
+
+  def setup
+    # setup runs at the beginning of each test
+    super
+    @acl_name = 'test-foo-1'
+    @seqno = 10
+    @afi = 'ip'
+    no_ip_access_list_foo
+  end
+
+  def teardown
+    # teardown runs at the end of each test
+    no_ip_access_list_foo
+    super
+  end
+
+  def no_ip_access_list_foo
+    # Turn the feature off for a clean test.
+    config('no ip access-list ' + @acl_name)
+  end
+
+  # TESTS
+
```

Comment:
Please add testing for the following.

1) ipv6
2) proto is an integer
3) 60 deny 55 6.6.6.6/24 addrgroup foo
4) 60 deny 55 6.6.6.6/24 addrgroup foo dscp cs2 fragments packet-length eq 30

Seems like there are even more combinations beyond this.